### PR TITLE
fix(anim_timeline) avoid calling lv_anim_del(NULL, NULL)

### PR DESCRIPTION
**Pull request by FASTSHIFT**
### Description of the feature or fix

This PR is about the resolution of https://github.com/lvgl/lvgl/issues/2589.
Avoid calling lv_anim_del(NULL, NULL) to cause all animations to be deleted.

### Checkpoints
- [x] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Update [CHANGELOG.md](https://github.com/lvgl/lvgl/blob/master/docs/CHANGELOG.md)
- [ ] Update the documentation

### CODE REVIEW

File: src/misc/lv_anim_timeline.c [hunk 1]
```diff
@@ -114,7 +114,11 @@ void lv_anim_timeline_stop(lv_anim_timeline_t * at)
 
     for(uint32_t i = 0; i < at->anim_dsc_cnt; i++) {
         lv_anim_t * a = &(at->anim_dsc[i].anim);
-        lv_anim_custom_del(a, (lv_anim_custom_exec_cb_t)a->exec_cb);
+
+        /*Avoid calling lv_anim_del(NULL, NULL) to cause all animations to be deleted*/
+        if (a->var != NULL || a->exec_cb != NULL) {
+            lv_anim_custom_del(a, (lv_anim_custom_exec_cb_t)a->exec_cb);
```
Comment (kisvegabor) [inline src/misc/lv_anim_timeline.c]: Why can't we use simple `lv_anim_del`?

Comment (FASTSHIFT) [reply src/misc/lv_anim_timeline.c]: OK

**Comment by kisvegabor** [comment]
So far I thought that an animation doesn't make much sens if neither `var` nor `exec_cb` is set. But setting only the `ready_cb` might be also valid use case, especially in case of anim timeline. 

An other solution could be to add a dummy callback to `exec_cb`  in `lv_anim_timeline_add` if it's `NULL`.

BTW, won't the current solution (in this PR) cause memory leak?

**Comment by FASTSHIFT** [comment]
> So far I thought that an animation doesn't make much sens if neither nor is set. But setting only the might be also valid use case, especially in case of anim timeline.`var``exec_cb``ready_cb`
> 
> An other solution could be to add a dummy callback to in if it's .`exec_cb``lv_anim_timeline_add``NULL`
> 
> BTW, won't the current solution (in this PR) cause memory leak?

Why is there a memory leak, is it `lv_anim_timeline` or `lv_anim`?

**Comment by kisvegabor** [comment]
> Why is there a memory leak, is it lv_anim_timeline or lv_anim?


Because if both `var` and `exec_cb` are `NULL` the animation won't be deleted.

**Comment by FASTSHIFT** [comment]
> > Why is there a memory leak, is it lv_anim_timeline or lv_anim?
> 
> Because if both `var` and `exec_cb` are `NULL` the animation won't be deleted.

`lv_anim` will automatically delete the animation when the animation reaches the set end time. No need to actively call `lv_anim_del` to delete.

**Comment by kisvegabor** [comment]
But if an animation is not deleted in `lv_anim_timeline_stop` it will keep running and it will call its `ready_cb`. However, the user intention was to make all animations stop.

**Comment by FASTSHIFT** [comment]
You are right. I thought of a solution: When it is detected that `var` and `exec_cb` are NULL, use the pointer of `lv_anim_timeline` as `var`, so that the animation can be completely deleted when `lv_anim_timeline_stop` is called.

**Comment by kisvegabor** [comment]
Looks good, thank you!

**Pull request merged**